django/db/models/aggregates.py

```diff
@@ -78,6 +78,14 @@ def __init__(self, expression, **extra):
         output_field = extra.pop('output_field', FloatField())
         super(Avg, self).__init__(expression, output_field=output_field, **extra)
 
+    def as_oracle(self, compiler, connection):
+        if self.output_field.get_internal_type() == 'DurationField':
```

Comment:
Aggregates/Expressions aren't registered against fields like lookups and transforms are, so that's not really an option. The backend could possibly register the as_oracle function against the aggregate via monkey patching, but I don't think we've landed on the perfect spot to do so, so that the registration definitely happens before the usage of the aggregate.

The `as_{vendor}` method of registering backend implementations is already being used in `models.db.functions`, so this isn't the first to do so.

I suppose we could look into changing how the `as_sql` method is resolved. Although the most common case so far of overriding the function is to provide a different `self.function`, rather than switching on the type of the output_field. I'd be interested in seeing what this could look like, but I think it's out of scope for this patch.